rust/agents/relayer/src/relayer.rs

```diff
@@ -114,129 +122,112 @@ impl BaseAgent for Relayer {
         let checkpoint_fetcher = CheckpointFetcher::new(
-            self.outbox().outbox(),
+            self.mailbox(&self.origin_chain_name).unwrap(),
             self.signed_checkpoint_polling_interval,
             self.multisig_checkpoint_syncer.clone(),
             signed_checkpoint_sender,
-            self.core.metrics.last_known_message_leaf_index(),
+            self.core.metrics.last_known_message_nonce(),
         );
         checkpoint_fetcher.spawn()
     }
 
-    fn run_outbox_metrics_loop(&self) -> Instrumented<JoinHandle<Result<()>>> {
-        let outbox = self.outbox().outbox().clone();
-        let outbox_name = outbox.chain_name();
-        let outbox_state_gauge = self
-            .core
-            .metrics
-            .outbox_state()
-            .with_label_values(&[outbox_name]);
-        tokio::spawn(async move {
-            let mut interval = tokio::time::interval(Duration::from_secs(60 * 10));
-            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
-            loop {
-                let state = outbox.state().await;
-                match &state {
-                    Ok(state) => outbox_state_gauge.set(*state as u8 as i64),
-                    Err(e) => tracing::warn!(error = %e, "Failed to get outbox state"),
-                };
-
-                interval.tick().await;
-            }
-        })
-        .instrument(info_span!("outbox_metrics_loop"))
-    }
-
     /// Helper to construct a new GelatoSubmitter instance for submission to a
-    /// particular inbox.
-    fn make_gelato_submitter_for_inbox(
+    /// particular mailbox.
+    fn make_gelato_submitter(
         &self,
         message_receiver: mpsc::UnboundedReceiver<SubmitMessageArgs>,
-        inbox_contracts: InboxContracts,
+        mailbox: CachingMailbox,
+        multisig_ism: CachingMultisigIsm,
         gelato_config: GelatoConf,
         gas_payment_enforcer: Arc<GasPaymentEnforcer>,
     ) -> GelatoSubmitter {
-        let inbox_chain_name = inbox_contracts.inbox.chain_name().to_owned();
+        let chain_name = mailbox.chain_name().to_owned();
         GelatoSubmitter::new(
             message_receiver,
-            inbox_contracts,
-            self.outbox().db().clone(),
+            mailbox,
+            multisig_ism,
+            self.mailbox(&self.origin_chain_name).unwrap().db().clone(),
             gelato_config,
-            GelatoSubmitterMetrics::new(
-                &self.core.metrics,
-                self.outbox().outbox().chain_name(),
-                &inbox_chain_name,
-            ),
+            GelatoSubmitterMetrics::new(&self.core.metrics, &self.origin_chain_name, &chain_name),
             gas_payment_enforcer,
         )
     }
 
-    #[tracing::instrument(fields(inbox=%inbox_contracts.inbox.chain_name()))]
-    fn run_inbox(
+    #[allow(clippy::too_many_arguments)]
+    #[tracing::instrument(fields(destination=%mailbox.chain_name()))]
+    fn run_mailbox(
```

Comment:
I find this name a little confusing when there's also `run_mailbox_sync`. I think it'd help to clarify in the name which is relevant for dispatched messages from an origin / for processing messages at a destination